src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/LinuxIcmpV6Statistics.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+
+namespace System.Net.NetworkInformation
+{
+    internal class LinuxIcmpV6Statistics : IcmpV6Statistics
+    {
+        private int _inDestUnreachs;
+        private int _outDestUnreachs;
+        private int _inEchoReplies;
+        private int _inEchos;
+        private int _outEchoReplies;
+        private int _outEchos;
+        private int _inErrors;
+        private int _outErrors;
+        private int _inGroupMembQueries;
+        private int _outInGroupMembQueries;
+        private int _inGroupMembReductions;
+        private int _outGroupMembReductions;
+        private int _inGroupMembResponses;
+        private int _outGroupMembResponses;
+        private int _inMsgs;
+        private int _outMsgs;
+        private int _inNeighborAdvertisements;
+        private int _outNeighborAdvertisements;
+        private int _inNeighborSolicits;
+        private int _outNeighborSolicits;
+        private int _inPktTooBigs;
+        private int _outPktTooBigs;
+        private int _inParmProblems;
+        private int _outParmProblems;
+        private int _inRedirects;
+        private int _outRedirects;
+        private int _inRouterSolicits;
+        private int _outRouterSolicits;
+        private int _inRouterAdvertisements;
+        private int _outRouterAdvertisements;
+        private int _inTimeExcds;
+        private int _outTimeExcds;
+
+        public static IcmpV6Statistics CreateIcmpV6Statistics()
+        {
+            LinuxIcmpV6Statistics stats = new LinuxIcmpV6Statistics();
+
```

Comment:
Same design observation: keep the implementation in the non-static portion and keep this as a factory method.